overlord/restart/restart_test.go

```diff
@@ -459,57 +477,93 @@ func (s *restartSuite) TestPendingForChange(c *C) {
 	_, err := restart.Manager(st, "boot-id-1", nil)
 	c.Assert(err, IsNil)
 
+	chg1 := st.NewChange("pending", "...")
+	chg1.Set("wait-for-system-restart", true)
+	t1 := st.NewTask("task", "...")
+	chg1.AddTask(t1)
+	t1.SetToWait(state.DoneStatus)
+	t1.Set("wait-for-system-restart-from-boot-id", "boot-id-1")
+
+	chg2 := st.NewChange("pending", "...")
+	chg2.Set("wait-for-system-restart", true)
+	t2 := st.NewTask("task", "...")
+	chg2.AddTask(t2)
+	t2.SetToWait(state.UndoneStatus)
+	t2.Set("wait-for-system-restart-from-boot-id", "boot-id-1")
+
+	c.Check(restart.PendingForChange(st, chg1), Equals, true)
+	c.Check(restart.PendingForChange(st, chg2), Equals, true)
+}
+
+func (s *restartSuite) TestPendingForChangeNoWaitTasks(c *C) {
+	st := state.New(nil)
+
+	st.Lock()
+	defer st.Unlock()
+
+	_, err := restart.Manager(st, "boot-id-1", nil)
+	c.Assert(err, IsNil)
+
 	chg1 := st.NewChange("not-ready", "...")
 	t1 := st.NewTask("task", "...")
 	chg1.AddTask(t1)
 	c.Check(chg1.IsReady(), Equals, false)
 
-	chg2 := st.NewChange("not-pending", "...")
-	t2 := st.NewTask("wait-task", "...")
-	t3 := st.NewTask("task", "...")
-	t4 := st.NewTask("task", "...")
-	chg2.AddTask(t2)
+	c.Check(restart.PendingForChange(st, chg1), Equals, false)
+}
+
+func (s *restartSuite) TestPendingForChangeWaitTasksButNotPending(c *C) {
+	release.MockOnClassic(false)
+	st := state.New(nil)
+
+	st.Lock()
+	defer st.Unlock()
+
+	_, err := restart.Manager(st, "boot-id-1", nil)
+	c.Assert(err, IsNil)
+
+	chg1 := st.NewChange("not-pending-do", "...")
+	t1 := st.NewTask("wait-task", "...")
+	t2 := st.NewTask("task", "...")
+	chg1.AddTask(t1)
+	chg1.AddTask(t2)
+	t2.WaitFor(t1)
+
+	err = restart.FinishTaskWithRestart(t1, state.DoneStatus, restart.RestartSystem, "some-snap", nil)
+	c.Assert(err, IsNil)
+	c.Check(t1.Status(), Equals, state.WaitStatus)
+	c.Check(t1.WaitedStatus(), Equals, state.DoneStatus)
+	c.Check(t2.Status(), Equals, state.DoStatus)
```

Comment:
maybe we can also add a check for pending change?
`
c.Check(restart.PendingForChange(st, chg1), Equals, true)
`